services/personal/api.go

```diff
@@ -29,15 +29,15 @@ type metadata struct {
 
 func newMetadata(rpcParams []interface{}) (*metadata, error) {
 	// personal_sign can be called with the following parameters
-	// 1) data to sign
-	// 2) account
-	// 3) (optional) password
+	//
+	// 1) account (from)
+	// 2) data to sign
 	// here, we always ignore (3) because we send a confirmation for the password to UI
-	if len(rpcParams) < 2 || len(rpcParams) > 3 {
+	if len(rpcParams) != 2 {
 		return nil, ErrSignInvalidNumberOfParameters
 	}
-	data := rpcParams[0]
-	address := rpcParams[1].(string)
+	address := rpcParams[0].(string)
```

Comment:
IMO this should be more defensive and do not panic if `rpcParams[0]` is not a string.